core/src/components/avatar/avatar.tsx

```diff
@@ -93,6 +98,7 @@ export class Avatar implements ComponentInterface {
           [`avatar-${shape}`]: shape !== undefined,
           [`avatar-image`]: this.hasImage,
           [`avatar-icon`]: this.hasIcon,
+          [`avatar-disabled`]: this.disabled,
```

Comment:
Nit: Might be worth doing something like
```typescript
const { disabled } = this;
[...]
[`avatar-disabled`]: disabled
```
You could do that with `hasImage` and `hasIcon` too and things might look a little more clean. This is just a nit though, no worries if you don't want to do that.